### DISCUSSION TITLE
Define owner of document in Appwrite

### DISCUSSION THREAD

**Discussion by netapy** [appwrite/appwrite#3773] [Q&A] [answered]
Hi, I'm currently building a social network type app, with Flutter and Appwrite.
I would like my users to be able to post Playlists that would be identified as theirs.

For this I have three fields, `name`, `music` and `owner` which identifies the user that created it.

Here is an example of the data structure on my "Playlist" collection : 
```
{
    "name": "My Awesome Playlist",
    "music": [],
    "owner": "bobby",
    "$read": [
        "role:all"
    ],
    "$write": [
        "user:bobby"
    ],
    "$id": "UUUUUUUUUUUUUUU",
    "$createdAt": 1662461698,
    "$updatedAt": 1662461698,
    "$collection": "XXXXXXXXXXXXXXX"
}
```

When a user navigates to another users' profile, I query and display all the Playlists for which  `user_id` == `owner`

The problem with this, is that anybody can then create a Playlist with a custom `owner`, by modifying the POST request and display content on other users' profiles.

➡️ **I'm sure this is not the best way to do things ; how would you recommend me to do this in Appwrite ?** 

(I would like to avoid using functions for verifications or custom requests.)

**Comment by stnguyen90** [comment, marked as answer]
@netapy, using a function is probably the only solution to let a user create a document but prevent them from changing something they shouldn't. I do [this for an app](https://github.com/stnguyen90/places/blob/f6589e1566d967f94ee9bcac941614da3dfa8192/appwrite-functions/create-comment/src/index.ts#L67) I made earlier this year.
